### DISCUSSION TITLE
Local background in sep.sumcircle

### DISCUSSION THREAD

**Discussion by aroy244** [sep-developers/sep#126] [Q&A]
Hi, I'm new to the Sep module. I have two questions:
1. In order to obtain the photometry using Sep.sum_circle, I'm using "bkgann = inner radius, outer radius" argument to subtract the local background from the data. Now my question is, what is this bkgann value? Is it mean, median or mode value of the specified annulus? 
Is there a way where I can specify the bkgann to take the median value from the specified annulus?  

2. The ** 'thresh' **  argument in sep.extract(data, thresh) is basically some number that is interpreted as that number \times error to detect the sources if I understand it correctly. Now, I'm not sure how the **error** is calculated. It will be of great help if someone explains it or send me some link where it's well explained. 

Thanks.

**Comment by kbarbary** [comment]
Hi @aroy244 ,

### Background annulus

The mean in the background annulus is used:  
> If supplied, the background is estimated by averaging unmasked pixels in this annulus.

(from https://sep.readthedocs.io/en/v1.0.x/api/sep.sum_circle.html). Internally, the code uses `sep.sum_circann` to sum up flux in the background annulus. See [here](https://github.com/kbarbary/sep/blob/master/sep.pyx#L998-L1016). Unfortunately, there's currently no option to use median or any other estimator.

### "thresh" and "err"

Regarding the `err` argument to `sep.extract`: A typical use is to set `err` to the standard deviation (or "rms") of the image. You can use a single value (in which case `err` is a single value) or a spatially variable estimate (in which case `err` is a 2-d array). The latter is useful when the noise varies across the image. The [tutorial](https://sep.readthedocs.io/en/v1.0.x/tutorial.html#Object-detection) has an example of using a single value:
```python
objects = sep.extract(data_sub, 1.5, err=bkg.globalrms)
```
One could instead use a spatially variable estimate of "rms" like so:
```python
objects = sep.extract(data_sub, 1.5, err=bkg.rms())
```
The tutorial has a bit more detail on using the `sep.Background()` class (the `bkg` object in the above examples).
